### PR TITLE
Docs site supports docs for unstable APIs

### PR DESCRIPTION
**Pull request by r100-stack**
## Changes

Moved some unstable API docs related changes from #2001 into this PR. This PR adds an Astro component called `UnstableApiCard` that is similar to the `Placeholder` but has a message that is instead the warning for the unstable API.

I thought to replace the info icon with a flask, similar to the experimental API icon in most other docs site. However, I couldn't find a flask icon. So, I used a warning icon for now.

Also, I added a `componentProp` in `PropsTable` to help choose the component who's props is to be displayed from the `.d.ts` file. (https://github.com/iTwin/iTwinUI/pull/2001#discussion_r1846639610)

## Testing

<details><summary>Old screenshot</summary>
<p>

image

</p>
</details> 

image

## Docs

N/A

### CODE REVIEW

File: apps/website/src/components/DocsNoteCard.astro [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+
+---
+
```
Comment (mayank99) [inline apps/website/src/components/DocsNoteCard.astro:4]: nit: frontmatter is not used

```suggestion
```

File: apps/website/src/components/UnstableApiCard.astro [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+import InformationIcon from '~/icons/InformationIcon.astro';
+import DocsNoteCard from './DocsNoteCard.astro';
+---
+
+<DocsNoteCard {...Astro.props}>
+  <InformationIcon aria-hidden='true' slot='icon' />
```
Comment (mayank99) [inline apps/website/src/components/UnstableApiCard.astro:7]: Maybe we should use a warning ⚠️ icon?

**Review by mayank99 — APPROVED** [review]

**Review by smmr-dn — APPROVED** [review]

File: apps/website/src/components/UnstableApiCard.astro [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+import WarningIcon from '~/icons/WarningIcon.astro';
+import DocsNoteCard from './DocsNoteCard.astro';
+---
+
+<DocsNoteCard {...Astro.props}>
+  <WarningIcon aria-hidden='true' slot='icon' />
+  <p slot='content'>
+    <strong>Unstable</strong>: This component is marked unstable to help us gather feedback and
```
Comment (mayank99) [inline apps/website/src/components/UnstableApiCard.astro:9]: I just realized that we are not actually asking for feedback.

Maybe we can create a "Feedback" category on GitHub Discussions, and link it here?

Comment (mayank99) [reply apps/website/src/components/UnstableApiCard.astro:9]: We could also repurpose the existing ["Documentation" cateogory](https://github.com/iTwin/iTwinUI/discussions/categories/documentation).

If we do this, will need to also update the `FeedbackLink.astro` component to point to the new category.

**Review by mayank99 — APPROVED** [review]

**Pull request merged**